### ISSUE TITLE
Cannot run Windows Terminal as the same user from two different sessions

### ISSUE THREAD

**Issue by conioh** [microsoft/terminal#17629]
### Windows Terminal version

1.20.11781.0

### Windows build number

10.0.22631.0

### Steps to reproduce

1. Log on to user _Admin_.
2. Launch Windows Terminal.
3. "Lock".
4. Log on to user _User_.
5. (Optional: Launch Windows Terminal and see that it works.)
6. Launch Windows Terminal as user _Admin_ (holding Ctrl+Shift while clicking on the taskbar icon, or use _Run as administrator_ from the Start menu, etc.) 

### Expected Behavior

Launch.

The same as would have happened if we had closed Windows Terminal is the first TS session.

### Actual Behavior

<img width="419" alt="Screenshot 2024-07-30 190000" src="https://github.com/user-attachments/assets/38f38bbd-ba5e-4929-90bf-0a86ece36cc2">

**Comment by carlos-zamora** [comment]
> This is not an inherent limitation of apps distributed via the Microsoft Store...

Good catch! This is on us. This won't be easy to fix, but we'll be sure to get to it. 😊

**Comment by zadjii-msft** [comment]
Other notes: This is probably because 
* the first user (Admin) instantiates the `Monarch`, 
* then the second user tries to instantiate one
* finds the first, and then can't actually instantiate it, because the server is running at a higher priv

but we can probably be smarter about the way we CoRegisterClassWhatever

**Comment by conioh** [comment]
> Other notes: This is probably because
> 
> * the first user (Admin) instantiates the `Monarch`,
> * then the second user tries to instantiate one
> * finds the first, and then can't actually instantiate it, because the server is running at a higher priv
> 
> but we can probably be smarter about the way we CoRegisterClassWhatever

Since we're running both processes as the same principal, how does that make sense?
Moreover, the issue also happens if both users are non-admins or if both users are admins. I've updated the issue to reflect that.
How can that be related to "running at a higher priv"? Could you explain?

**Comment by DHowett** [comment]
Sorry, I think Mike misspoke. It's not about privileges or integrity levels, but that the application is running in a different session, but it has a live registration which is inaccessible to the new instance.

**Comment by conioh** [comment]
> Sorry, I think Mike misspoke. It's not about privileges or integrity levels, but that the application is running in a different session, but it has a live registration which is inaccessible to the new instance.

Do you mean a registration with this Monarch he mentioned?

I found a bunch of references to this in the code (and also to Emperors and Peasants) but not sure what that it. Based on comments such as [this](https://github.com/microsoft/terminal/blob/a8582978afa50ece88edc7eda9e182ced64876e2/src/cascadia/TerminalApp/TerminalPage.cpp#L2232) and [this](https://github.com/microsoft/terminal/blob/a8582978afa50ece88edc7eda9e182ced64876e2/src/cascadia/WindowsTerminal/WindowEmperor.cpp#L310-L320) I'm guessing it's something like an entity that functions as the core of a "main process" and manages all the windows (including windows created by subsequent processes, similar to how in Chromium based browsers when you run a second+ instance of `{chromium}.exe` it messages the "main" process to create a window and exits).

Assuming that's more or less what it is, I think it's not related. Perhaps it would also prove to be a problem, but the current problems happens before we get to that part.

1. This limitation exists on _many_ Store apps, but not on _all_ of them. This strongly indicates that it's _related_ to being to Store app, but not a _necessary_ consequence of it.
2. The issue doesn't happen when both sessions "belong" to the same user. In this case there are two `WindowsTerminal.exe` processes, each with his own Monarch in his own session. For example:
   <img width="229" alt="image" src="https://github.com/user-attachments/assets/11bbfcce-5c2b-4108-9730-9213da4bce36">
   You can see here WT with 2 tabs open in session 2 and another instance of WT with a single tab running in session 3. Both are running as the same user.
   This demonstrates that two Monarchs can live side by side, in two different TS sessions, running as the same user.
3. ProcMon on trying to run WT that way shows an error trying to access some file in the "Windows side of things" (rather than in something that WT did).

This leads me to suspect something like: WT, in it's manifest, does something similar to what _many but not all_ other Store apps (e.g. Sysinternals) do, while WinDbg does something different that allows this to work.

But whatever this Monarch is, I'm pretty sure _this_ error is not because of it. Maybe once we discover the offending configuration in the manifest and change it we'll discover another problem stemming from this Monarch thing, but I don't think this is it now.
